### PR TITLE
change: add request-validation plugin input params schema valid.

### LINKED ISSUE #1881: request-validation: check whether the input configuration is a valid JSON Schema.
https://github.com/apache/incubator-apisix/pull/1709#discussion_r440599144

### PR DESCRIPTION
**Pull request by shuaijinchao**
### What this PR does / why we need it:
FIX #1881 

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Is this PR backward compatible?

### CODE REVIEW

File: apisix/core/schema.lua [hunk 1]
```diff
@@ -36,16 +36,27 @@ local function create_validator(schema)
...
 end
 
-
-function _M.check(schema, json)
+local function get_validator(schema)
     local validator, err = cached_validator(schema, nil,
                                 create_validator, schema)
+
+    if not validator then
+        return false, err
```
Comment (membphis) [inline apisix/core/schema.lua:44]: `false` -> `nil` is better

File: apisix/plugins/request-validation.lua [hunk 2]
```diff
@@ -42,7 +42,28 @@ local _M = {
 
 
 function _M.check_schema(conf)
-    return core.schema.check(schema, conf)
+    local ok, err = core.schema.check(schema, conf)
+    if not ok then
+        return nil, err
+    end
+
+    if conf.body_schema then
```
Comment (membphis) [inline apisix/plugins/request-validation.lua:50]: this style is better.

```lua
if conf.body_schema then
    local ok, err = ....
    if not ok then
        return ....
    end
end
```

File: apisix/plugins/request-validation.lua [hunk 3]
```diff
@@ -42,7 +42,28 @@ local _M = {
...
+        ok, err = core.schema.valid(conf.body_schema)
+    end
+
+    if not ok then
+        return nil, err
```
Comment (membphis) [inline apisix/plugins/request-validation.lua:55]: we should use `false` here

**Comment by shuaijinchao** [comment]
@membphis already fixed, please recheck.

**Comment by membphis** [comment]
@shuaijinchao merged, many thx

**Pull request merged**